kubernetes-extension/src/main/java/org/ballerinax/kubernetes/handlers/istio/IstioGatewayHandler.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinax.kubernetes.handlers.istio;
+
+import org.ballerinax.kubernetes.exceptions.KubernetesPluginException;
+import org.ballerinax.kubernetes.handlers.AbstractArtifactHandler;
+import org.ballerinax.kubernetes.models.istio.IstioGatewayModel;
+import org.ballerinax.kubernetes.models.istio.IstioServerModel;
+import org.ballerinax.kubernetes.utils.KubernetesUtils;
+import org.yaml.snakeyaml.DumperOptions;
+import org.yaml.snakeyaml.Yaml;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import static org.ballerinax.kubernetes.KubernetesConstants.ISTIO_GATEWAY_FILE_POSTFIX;
+import static org.ballerinax.kubernetes.KubernetesConstants.YAML;
+
+/**
+ * Generates istio gateway artifacts.
+ *
+ * @since 0.983.0
+ */
+public class IstioGatewayHandler extends AbstractArtifactHandler {
+    
+    /**
+     * {@inheritDoc}
+     * Performs validations and creates the artifacts.
+     */
+    @Override
+    public void createArtifacts() throws KubernetesPluginException {
+        Map<String, IstioGatewayModel> gatewayModels = dataHolder.getIstioGatewayModels();
+        int size = gatewayModels.size();
+        if (size > 0) {
+            OUT.println();
+        }
+        
+        int count = 0;
+        for (IstioGatewayModel gatewayModel : gatewayModels.values()) {
+            count++;
+            
+            // Validate number of selectors.
+            if (null == gatewayModel.getSelector() || gatewayModel.getSelector().size() == 0) {
+                Map<String, String> selectors = new LinkedHashMap<>();
+                selectors.put("istio", "ingressgateway");
```

Comment:
Let's move the selector key to a string constant.

Reply:
Fixed with https://github.com/ballerinax/kubernetes/pull/220/commits/5f8887c1388816ebe18d4066c627a7d43a88bea9